Add authoring support for `@TitleHeading` directive (#611)
apple/swift-docc: Sources/docc/DocCDocumentation.docc/Reference Syntax/API Reference Syntax/TitleHeading.md

# ``docc/TitleHeading``

> Configures a symbol's documentation page and any references to that page to show a custom title heading instead of the default (page kind).

Reviewer: In this case, unlike DisplayName, this won't affect references to the symbols since the TitleHeading is only rendered on the page itself. Maybe something like this to align with [PageColor](https://www.swift.org/documentation/docc/pagecolor)?

```suggestion
A directive that specifies a title heading for a given documentation page.
```

It's a little vague but I think your overview does a good job of expanding on it.

@Metadata {
    @DocumentationExtension(mergeBehavior: override)
}

- Parameters:

>     - headingText: The text for the custom title heading.

Reviewer: ```suggestion
    - heading: The text for the custom title heading.
```

(If we update the property name above, we'll also need to update here.)
